terminal/bundles/org.eclipse.terminal.view.ui/src/org/eclipse/terminal/view/ui/TerminalViewId.java

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ * Copyright (c) 2011, 2025 Wind River Systems, Inc. and others. All rights reserved.
+ * This program and the accompanying materials are made available under the terms
+ * of the Eclipse Public License 2.0 which accompanies this distribution, and is
+ * available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * Wind River Systems - initial API and implementation
+ * Max Weninger (Wind River) - [361363] [TERMINALS] Implement "Pin&Clone" for the "Terminals" view
+ * Alexander Fedorov (ArSysOp) - further evolution
+ *******************************************************************************/
+package org.eclipse.terminal.view.ui;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.ui.IViewReference;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+
+/**
+ * Encapsulates the pair of mandatory primary terminal view id and optional secondary id
+ *
+ * @see IUIConstants#ID
+ * @see IViewReference#getSecondaryId()
+ */
+public final class TerminalViewId {
+
+	private final String primary;
+	private final Optional<String> secondary;
+
+	public TerminalViewId() {
+		this(IUIConstants.ID);
+	}
+
+	public TerminalViewId(String primary) {
+		this(primary, Optional.empty());
+	}
+
+	public TerminalViewId(String primary, String secondary) {
+		this(primary, Optional.ofNullable(secondary));
+	}
+
+	public TerminalViewId(String primary, Optional<String> secondary) {
+		this.primary = Objects.requireNonNull(primary);
+		this.secondary = Objects.requireNonNull(secondary);
+	}
+
+	public String primary() {
+		return primary;
+	}
+
+	public Optional<String> secondary() {
+		return secondary;
+	}
+
+	public TerminalViewId next() {
+		return new TerminalViewId(primary, nextSecondaryId());
+	}
```

Comment:
All call-sites of next() seem to perform `new TerminalId().next()` so it this really suitable here as an instance method and should not be more a static method like


```suggestion
	public static TerminalViewId create() {
		return new TerminalViewId(IUIConstants.ID, nextSecondaryId(IUConstants.ID));
	}
```

Reply:
I'm not so sure regarding this. 
We allow to specify primary id for some reason, perhaps we expect terminal view with another view id to use this API (may be similar with what we have for "Problem" view family).